xdsl/dialects/llvm.py

```diff
@@ -144,6 +144,41 @@ def from_size_and_type(size: int | IntAttr, type: Attribute):
         return LLVMArrayType([size, type])
 
 
+@irdl_attr_definition
+class LinkageAttr(ParametrizedAttribute):
+    name = "llvm.linkage"
+
+    linkage: ParameterDef[StringAttr]
+
+    def print_parameters(self, printer: Printer) -> None:
+        printer.print_string("<")
+        printer.print_attribute(self.linkage)
+        printer.print_string(">")
+
+    @staticmethod
+    def parse_parameters(parser: BaseParser) -> list[Attribute]:
+        if not parser.tokenizer.starts_with('<'):
+            return [NoneAttr()]
+        parser.parse_characters('<', "llvm.linkage parameter expected")
+        # The linkage string is output from xDSL as a string (and accepted by MLIR as such)
+        # however it is always output from MLIR without quotes. Therefore need to determine
+        # whether this is a string or not and slightly change how we parse based upon that
+        linkage_str = parser.try_parse_string_literal()
+        if linkage_str is not None:
+            linkage_str = linkage_str.string_contents
+        else:
+            linkage_str = parser.tokenizer.next_token().text
+        linkage = StringAttr(linkage_str)
+        parser.parse_characters('>', "End of llvm.linkage parameter expected!")
+        return [linkage]
+
+    @staticmethod
+    def from_linkage(linkage: str | StringAttr):
+        if isinstance(linkage, str):
+            linkage = StringAttr(linkage)
+        return LinkageAttr([linkage])
+
```

Comment:
Would it make sense to add a verifier checking that the string match an actual LLVM linkage value? AFAIU this currently allows to set any arbitrary string as a linkage.

Reply:
Good idea, I have added a verify method now